### PR TITLE
LPD-24865 - Add boilerplate project for using node build scripts directly in portal

### PR DESCRIPTION
**Pull request by bryceosterhaus**
https://liferay.atlassian.net/browse/LPD-24864

The long term goal is to move things from liferay-npm-scripts directly into liferay-portal and no longer have to import a separate npm package.

This npm project in liferay-portal allows us to call `yarn run node-scripts foo bar baz` from any module within portal without having to rely on an external npm package(`@liferay/npm-scripts`).

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:relevant

File: modules/node-scripts/bin.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env node
```
Comment (izaera) [inline modules/node-scripts/bin.js:1]: I think this should use the node binary that gradle downloads for us...

File: modules/node-scripts/bin.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {main} from './index.js';
+
+main().catch((error) => {
```
Comment (izaera) [inline modules/node-scripts/bin.js:9]: You can change this to:

```javascript
try {
  await main();
}
catch(error) {
  ...
}
```

For better readibility. Apparently you can use top level await in node :tada: .

**Comment by bryceosterhaus** [comment]
ci:stop

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:2565442150

**Comment by bryceosterhaus** [comment]
ci:forward:force

**Pull request closed without merging**